metadata/src/test/java/org/apache/kafka/controller/PartitionChangeBuilderTest.java

```diff
@@ -125,7 +125,7 @@ private static MetadataVersion metadataVersionForPartitionChangeRecordVersion(sh
             case (short) 1:
                 return MetadataVersion.IBP_3_7_IV2;
             case (short) 2:
-                return MetadataVersion.IBP_3_8_IV0;
+                return MetadataVersion.IBP_3_9_IV1;
```

Comment:
Should we add 3.8-IV0 to the following? The intention seems to be testing the latest production MV for each minor release. Then, I don't understand why we include 3.6-IV0 instead of 3.6-IV2. Is there a way to avoid manually add the latest production MV in the future?

```
    @ValueSource(strings = {"3.6-IV0", "3.7-IV4"})
    public void testNoLeaderEpochBumpOnIsrShrink(String metadataVersionString) {
```

```
    @ValueSource(strings = {"3.6-IV0", "3.7-IV4"})
    public void testLeaderEpochBumpOnIsrShrinkWithZkMigration(String metadataVersionString) {
```

```
    @ValueSource(strings = {"3.4-IV0", "3.5-IV2", "3.6-IV0", "3.7-IV4"})
    public void testNoLeaderEpochBumpOnIsrExpansion(String metadataVersionString) {
```

```
    @ValueSource(strings = {"3.4-IV0", "3.5-IV2", "3.6-IV0", "3.7-IV4"})
    public void testNoLeaderEpochBumpOnIsrExpansionDuringMigration(String metadataVersionString) {
```

```
    @ValueSource(strings = {"3.4-IV0", "3.5-IV2", "3.6-IV0", "3.7-IV4"})
    public void testLeaderEpochBumpOnNewReplicaSetDisjoint(String metadataVersionString) {
```

```
    @ValueSource(strings = {"3.4-IV0", "3.5-IV2", "3.6-IV0", "3.7-IV4"})
    public void testNoLeaderEpochBumpOnEmptyTargetIsr(String metadataVersionString) {
```

Reply:
`3.8-IV0` doesn't need to be added here since it doesn't change the version of `PartitionChangeRecord` that we generate.

IBP_3_7_IV2 introduced PartitionChangeRecord v1.

IBP_3_9_IV1 will introduce PartitionChangeRecord v2.

Reply:
Ok. In that case, should we change 3.7-IV4 to 3.7-IV2? Currently, it's not clear why 3.7-IV4 is chosen.

Also, it seems that IBP_3_9_IV1 has already introduced PartitionChangeRecord v2. Should we add IBP_3_9_IV1 in all those tests?

Reply:
> Ok. In that case, should we change 3.7-IV4 to 3.7-IV2? Currently, it's not clear why 3.7-IV4 is chosen.

OK. I will change the occurrences of `3.7-IV4` to `3.7-IV2`

> Also, it seems that IBP_3_9_IV1 has already introduced PartitionChangeRecord v2. Should we add IBP_3_9_IV1 in all those tests?

Fair point. `3.9-IV1` is already being tested in many places, but I will add it to a few more that I can see using `PartitionChangeRecord`.